### ISSUE TITLE
Nexus should store more data on Persistent Volume

### ISSUE THREAD

**Issue by novakg** [fabric8io/fabric8#6688]
Nexus instance uses a Persistent Volume to store its /sonatype-work/storage directory on it. This is very useful as it can also be used as a backup, so in case F8 crashes and must be re-installed, it can re-use the volume. However having the storage directory only does not contain all the data needed for full restore. At least the /sonatype-work/conf dir should be stored on PV. Might be the best to have the whole /sonatype-work directory on the PV, although I don't know if a sudden crash of Nexus that would leave half-done things there could be recovered from the whole dir, or it would be better to have the repository data and the config only and let Nexus re-build all the rest.

**Comment by rawlingsj** [comment]
We looked at mounting `/sonatype-work` before and we ran into an issue, I think it was something to do with nexus loosing the default repositories that are shipped in the docker image.  

We do need to find a solution for this and other too such as jenkins as that suffers from the same problem.

**Comment by rawlingsj** [comment]
Good point!  Yeah we can add that in.  Everyone's got quite a few things on at the moment but you could have a go and contribute back if you fancy it?

You'd need to fork and clone https://github.com/fabric8io/fabric8-devops, we use the fabric8 maven plugin https://maven.fabric8.io/ to generate all the yaml resources so you can `cd fabric8-devops/fabric8-docker-registry` and find the kubernetes / openshift fragments under `src/main/fabric8`.  

Here's an [example PVC](https://github.com/fabric8io/fabric8-devops/blob/master/nexus/src/main/fabric8/nexus-data-pvc.yml) you can copy into `src/main/fabric8` and [example matching volume mounts](https://github.com/fabric8io/fabric8-devops/blob/master/nexus/src/main/fabric8/nexus-deployment.yml#L61-L68) you'd need to add to the https://github.com/fabric8io/fabric8-devops/blob/master/fabric8-docker-registry/src/main/fabric8/fabric8-docker-registry-deployment.yml#L27

To test we tend to use [minikube locally](https://fabric8.io/guide/getStarted/gofabric8.html), then to build and apply your changes use..
```
cd fabric8-devops/fabric8-docker-registry
mvn clean install fabric8:deploy
```
We mainly hang on on irc.freenode.net #fabric8 or a few other ways if you want to chat about it.. https://fabric8.io/community/index.html

**Comment by novakg** [comment]
Thanks for the detailed description about how to do it. I give it a go. 

Just one more question for now: should I follow the steps described at http://fabric8.io/contributing/index.html to commit back my changes once I'm done?

**Comment by rawlingsj** [comment]
Yeah apart from the formatting that link is the best description of contributing changes.  I can see from your pull request that you figured that out though.  

It's great to see contributions BTW - thank you!
